src/Accelerators/NNPA/Transform/ZLow/ZLowRewrite.cpp

```diff
@@ -0,0 +1,119 @@
+//===-------- ZLowRewrite.cpp - ZLow Rewrite Patterns ---------------------===//
+//
+// Copyright 2022 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This pass implements optimizations for for ZLow operations.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Interfaces/ViewLikeInterface.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/DialectConversion.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+#include "src/Accelerators/NNPA/Dialect/ZLow/ZLowOps.hpp"
+#include "src/Accelerators/NNPA/Pass/DLCPasses.hpp"
+
+using namespace mlir;
+
+/// This pattern rewrites
+/// ```mlir
+///   zlow.unstick(%input, %output)
+///   %view = viewOp(%output)
+///   zlow.stick(%view, %res)
+/// ```
+/// by removing `zlow.stick` and replacing `%res` by `%input`, which is
+/// constrained by that `%input` and `%res` have the same static shape.
+/// This pattern potentially removes `zlow.unstick` and `viewOp` if they are
+/// dangling.
+///
+/// `viewOp` can be any op that inherits ViewLikeOpInterface, e.g.
+/// memref.reinterpret_cast, memref.collapse_shape, memref.expand_shape.
+//
+class StickViewUnstickRemovalPattern : public OpRewritePattern<ZLowStickOp> {
```

Comment:
Suggest to derive from `ConversionPattern`, are there any advantages using  an OpRewritePattern ?

Reply:
By using OpRewritePattern we can use a derived operation instead of raw operation as in ConversionPattern. As you see, we have ZLowStickOp here instead of Operation, so we don't need to create an operator adaptor to access ZLowStickOp's members.